app/experimenter/nimbus-ui/src/components/TableMetricPrimary/index.tsx

```diff
@@ -3,9 +3,88 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React from "react";
+import { AnalysisData } from "../../lib/visualization/types";
+import TableVisualizationRow from "../TableVisualizationRow";
+import { getExperiment_experimentBySlug_primaryProbeSets } from "../../types/getExperiment";
+import {
+  PRIMARY_METRIC_COLUMNS,
+  TABLE_LABEL,
+  DISPLAY_TYPE,
+} from "../../lib/visualization/constants";
 
-const TableMetricPrimary: React.FunctionComponent = () => {
-  return <div data-testid="table-metric-primary">primary metric table</div>;
+type PrimaryMetricStatistic = {
+  name: string;
+  displayType: DISPLAY_TYPE;
+  branchComparison?: string;
+  value?: string;
+};
+
+type TableMetricPrimaryProps = {
+  results: AnalysisData["overall"];
+  probeSet: getExperiment_experimentBySlug_primaryProbeSets | null;
+};
+
+const getStatistics = (
+  probeset: string | null,
+): Array<PrimaryMetricStatistic> => {
+  const probesetMetricID = `${probeset}_ever_used`;
+
+  // Make a copy of `PRIMARY_METRIC_COLUMNS` since we modify it.
+  const primaryMetricStatisticsList = PRIMARY_METRIC_COLUMNS.map(
+    (statistic: PrimaryMetricStatistic) => {
+      statistic["value"] = probesetMetricID;
+      return statistic;
+    },
+  );
+
+  return primaryMetricStatisticsList;
+};
+
+const TableMetricPrimary = ({
+  results = {},
+  probeSet,
+}: TableMetricPrimaryProps) => {
+  const primaryMetricStatistics = getStatistics(probeSet!.slug);
+  const metricKey = `${probeSet?.slug}_ever_used`;
+
+  return (
+    <div data-testid="table-metric-primary">
+      <h2 className="h5 my-3">{probeSet?.name}</h2>
```

Comment:
In my branch, I changed these classes to `h5 mb-3 mt-4` because I thought it needed a tiny bit more spacing up above, you could update here for consistency :)